tests/spec/w3c/headers-spec.js

```diff
@@ -557,37 +557,156 @@ describe("W3C — Headers", () => {
+
+      const h1 = doc.querySelector("h1#title");
+      expect(h1).toBeTruthy();
+      expect(h1.innerHTML).toBe("This should be <code>pass</code>.");
     });
 
-    it("uses <h1> if already present", async () => {
+    it("uses h1#title content when h1#title has content", async () => {
       const ops = makeStandardOps();
-      ops.body = `<h1 id='title'><code>pass</code></h1>${makeDefaultBody()}`;
+      ops.body = `
+        <h1 id='title'><code>pass</code></h1>
+        ${makeDefaultBody()}
+      `;
       const doc = await makeRSDoc(ops);
 
       // Title was relocated to head
       const titleInHead = doc.querySelector(".head h1");
-      expect(titleInHead.classList).toContain("p-name");
       expect(titleInHead.id).toBe("title");
 
       // html is not escaped
-      expect(titleInHead.firstChild.tagName).toBe("CODE");
+      expect(titleInHead.firstChild.localName).toBe("code");
       expect(titleInHead.textContent).toBe("pass");
+    });
+
+    it("makes h1 win even when h1#title is only whitespace", async () => {
+      const body = `
+        <h1 id='title'>       </h1>
+        ${makeDefaultBody()}
+      `;
+      const ops = makeStandardOps({}, body);
+      const doc = await makeRSDoc(ops);
+      expect(doc.title).toBe("");
+
+      const titleElem = doc.querySelector("title");
+      expect(titleElem).toBeTruthy();
+      expect(titleElem.textContent).toBe("");
+
+      const h1 = doc.querySelector("h1#title");
+      expect(h1).toBeTruthy();
+      expect(h1.classList).toContain("respec-offending-element");
+    });
+
+    it('makes h1 always win even when h1#title is ""', async () => {
+      const body = `
+      <h1 id='title'></h1>${makeDefaultBody()}`;
+      const ops = makeStandardOps({}, body);
+      const doc = await makeRSDoc(ops);
+      expect(doc.title).toBe("");
+
+      const titleElem = doc.querySelector("title");
+      expect(titleElem).toBeTruthy();
+      expect(titleElem.textContent).toBe("");
+
+      const h1 = doc.querySelector("h1#title");
+      expect(h1).toBeTruthy();
+      expect(h1.textContent).toBe("");
+      expect(h1.classList).toContain("respec-offending-element");
+    });
+  });
+
+  describe("precedence rules for title when h1#title element is not present", () => {
+    it("uses <title> when it contains a non-empty string", async () => {
+      const body = `<title>Title!!!</title>${makeDefaultBody()}`;
+      const ops = makeStandardOps({}, body);
+      const doc = await makeRSDoc(ops);
+      expect(doc.title).toBe("Title!!!");
+
+      const titleElem = doc.querySelector("title");
+      expect(titleElem).toBeTruthy();
+      expect(titleElem.textContent).toBe("Title!!!");
+
+      const h1 = doc.querySelector("h1#title");
+      expect(h1).toBeTruthy();
+      expect(h1.textContent).toBe("Title!!!");
+    });
+
+    it("adds h1#title with correct text and uses a default title when the document excludes a title", async () => {
```

Comment:
```suggestion
    it("uses a default title when the document excludes a title", async () => {
```

The "correct text" is a default title, so it seems a bit redundant 👀